ymer/src/main/java/com/avanza/ymer/MongoDocumentCollection.java

```diff
@@ -118,12 +141,25 @@ public void update(Document newVersion) {
 	}
 
 	@Override
-	public void updateById(Object id, Map<String, Object> fieldsAndValuesToSet) {
-		fieldsAndValuesToSet.entrySet().stream().map(entry -> Updates.set(entry.getKey(), entry.getValue())).reduce(Updates::combine)
-				.ifPresent(updates -> {
-					UpdateResult updateResult = collection.updateOne(Filters.eq(id), updates);
-					idValidator.validateUpdatedExistingDocument("update", updateResult, new Document("_id", id));
-				});
+	public void bulkWrite(Consumer<BulkWriter> bulkWriter) {
+		List<WriteModel<Document>> writeModels = new ArrayList<>();
+		bulkWriter.accept((ids, fieldsToSet) -> {
+			Bson updates = toUpdates(fieldsToSet);
+			if (ids.isEmpty()) {
+				log.warn("Skipping updates because no ids provided");
```

Comment:
Maybe lower the severity of this logging? This gets logged a lot now when running on a db whose `_instanceId` values are already correct. Or an empty-check when this is called from `PersistedInstanceIdRecalculationService`?

Reply:
I added an empty check to `PersistedInstanceIdRecalculationService` in https://github.com/AvanzaBank/ymer/pull/62/commits/8861749427d3be38f305897cd7c884fc0c40eec7